src/data/h1/index.js

```diff
@@ -0,0 +1,50 @@
+const basicTagsList = require("./tags");
+const {getDirectReferencedTagNames} = require("./invader");
+
+const basicTagsById = Object.fromEntries(basicTagsList.map(basicTag =>
+  [basicTag.id, basicTag]
+));
+
+function buildData(invaderStructDefs) {
+  //augment the basic list of tags with more detail provided by external libs
+  const tags = basicTagsList.map(basicTag => ({
+    ...basicTag,
+    invaderStruct: basicTag.invaderStructName ?
+      invaderStructDefs[basicTag.invaderStructName] :
+      null
+  }));
+
+  const tagsById = Object.fromEntries(tags.map(tag =>
+    [tag.id, tag]
+  ));
+  const tagsByName = Object.fromEntries(tags.map(tag =>
+    [tag.name, tag]
+  ));
+
+  //next, build most inter-tag relationships
+  for (let tag of tags) {
+    tag.references = [];
+    if (tag.invaderStructName) {
```

Comment:
Not all tags have corresponding invader structs (e.g. UI widgets, and unused tags like spheroid).

When an invader struct is available, it's used to compute tag reference information. It can also be used later to display a table of tag fields within the article.